app/models/package_manager/nu_get.rb

```diff
@@ -132,34 +133,44 @@ def self.get_releases(name)
     end
 
     def self.mapping(raw_project)
-      item = raw_project[:releases].last
-      raw_nuspec = nuspec(raw_project[:name], item.version_number)
+      latest_raw_version = raw_project[:raw_versions].last
+      raw_nuspec = nuspec(raw_project[:name], latest_raw_version.version_number)
       nuspec_repo = raw_nuspec&.locate("package/metadata/repository")&.first
       nuspec_repo = nuspec_repo["url"] if nuspec_repo
 
       MappingBuilder.build_hash(
         name: raw_project[:name],
-        description: item.description,
-        homepage: item.project_url,
-        keywords_array: item.tags,
-        repository_url: repo_fallback(nuspec_repo, item.project_url),
-        licenses: item.licenses,
+        description: latest_raw_version.description,
+        homepage: latest_raw_version.project_url,
+        keywords_array: latest_raw_version.tags,
+        repository_url: repo_fallback(nuspec_repo, latest_raw_version.project_url),
+        licenses: latest_raw_version.licenses,
         versions: versions(raw_project, raw_project[:name])
       )
     end
 
     def self.versions(raw_project, _name)
-      raw_project[:releases].map do |item|
-        VersionBuilder.build_hash(
-          number: item.version_number,
-          published_at: item.published_at,
-          original_license: item.original_license
-        )
+      raw_project[:raw_versions].map do |raw_version|
+        version_data = {
+          number: raw_version.version_number,
+          published_at: raw_version.published_at,
+          original_license: raw_version.original_license,
+        }
+
+        if raw_version.deprecation.present?
+          # We considered NuGet versions that are unlisted as "Deprecated", and those
+          # versions have their "published" reset to 1/1/1900 in the NuGet API.
+          # Keep the original "published_at" on our side but mark it as deprecated.
+          version_data.delete(:published_at)
+          version_data[:status] = "Deprecated"
+        end
```

Comment:
@mpace965 re: [this discussion](https://github.com/librariesio/libraries.io/pull/3397#discussion_r1624786708), I forgot to mention that we'll want to avoid setting `published_at: nil` in this change. Instead we're keeping the original "published_at" value instead of setting to 1900 or to nil.